tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandlerWithAuthorizationTest.java

```diff
@@ -0,0 +1,245 @@
+        handler.ctx = mockCtx;
+    }
+
+    @Override
+    protected void createAdmin() throws Exception {
+        super.admin = spy(PulsarAdmin.builder().serviceHttpUrl(brokerUrl.toString())
+                .authentication(this.conf.getBrokerClientAuthenticationPlugin(),
+                        this.conf.getBrokerClientAuthenticationParameters()).build());
+    }
+
+    @AfterClass
+    @Override
+    protected void cleanup() throws Exception {
+        adminManager.shutdown();
+        super.internalCleanup();
+    }
+
+    @DataProvider(name = "metadataVersions")
+    public static Object[][] metadataVersions() {
+        return new Object[][]{ { (short) 0 }, { (short) 1 } };
+    }
+
+    @Test(timeOut = 10000, dataProvider = "metadataVersions")
+    public void testMetadataForPartitionedTopicFailed(short version) throws Exception {
+        final String topic = "testMetadataForPartitionedTopic-" + version;
+        admin.topics().createNonPartitionedTopic("persistent://" + TENANT + "/" + NAMESPACE + "/" + topic);
+
+        final RequestHeader header = new RequestHeader(ApiKeys.METADATA, version, "client", 0);
+        final MetadataRequest request =
+                new MetadataRequest(Collections.singletonList(topic), false, version);
+        final CompletableFuture<AbstractResponse> responseFuture = new CompletableFuture<>();
+        handler.handleTopicMetadataRequest(
+                new KafkaCommandDecoder.KafkaHeaderAndRequest(
+                        header, request, PulsarByteBufAllocator.DEFAULT.heapBuffer(), null),
+                responseFuture);
+        final MetadataResponse response = (MetadataResponse) responseFuture.get();
+        assertEquals(response.topicMetadata().size(), 1);
+        assertEquals(response.errors().size(), 1);
+        assertEquals(response.errors().get(topic), Errors.TOPIC_AUTHORIZATION_FAILED);
+    }
+
+    @Test(timeOut = 10000, dataProvider = "metadataVersions")
+    public void testMetadataForPartitionedTopicSuccess(short version) throws Exception {
+        final String topic = TOPIC + "-" + version;
+        KafkaRequestHandler spyHandler = spy(handler);
+        doReturn(CompletableFuture.completedFuture(true))
+                .when(spyHandler)
+                .authorize(eq(AclOperation.DESCRIBE), eq(Resource.of(ResourceType.TOPIC, topic)));
+        final RequestHeader header = new RequestHeader(ApiKeys.METADATA, version, "client", 0);
+        final MetadataRequest request =
+                new MetadataRequest(Collections.singletonList(topic), true, version);
+        final CompletableFuture<AbstractResponse> responseFuture = new CompletableFuture<>();
+        spyHandler.handleTopicMetadataRequest(
+                new KafkaCommandDecoder.KafkaHeaderAndRequest(
+                        header, request, PulsarByteBufAllocator.DEFAULT.heapBuffer(), null),
+                responseFuture);
+         final MetadataResponse response = (MetadataResponse) responseFuture.get();
```

Comment:
This line is not aligned, but I don't know why checkstyle plugin doesn't point out this.

Reply:
Yes, it's strange.